linenumberarea.h

```diff
@@ -0,0 +1,21 @@
+#ifndef LINENUMBERAREA_H
+#define LINENUMBERAREA_H
+
+#include <QWidget>
+#include "codeeditor.h"
+
+class LineNumberArea : public QWidget
+{
+public:
+    explicit LineNumberArea(CodeEditor *editor);
```

Comment:
It's  recommended that we have a short description for each function, especially for the public interface.